rust/load/src/lib.rs

```diff
@@ -255,22 +266,40 @@ impl Workload {
             }
             Workload::Hybrid(hybrid) => {
                 let scale: f64 = any(guac);
-                let mut total = scale * hybrid.iter().map(|(p, _)| *p).sum::<f64>();
+                let mut total = scale
+                    * hybrid
+                        .iter()
+                        .filter_map(|(p, w)| if w.is_active() { Some(*p) } else { None })
+                        .sum::<f64>();
                 for (p, workload) in hybrid {
                     if *p < 0.0 {
                         return Err(Box::new(Error::InvalidRequest(
                             "hybrid probabilities must be positive".to_string(),
                         )));
                     }
-                    if *p >= total {
-                        return Box::pin(workload.step(client, data_set, guac)).await;
+                    if workload.is_active() {
+                        if *p >= total {
+                            return Box::pin(workload.step(client, data_set, guac)).await;
+                        }
+                        total -= *p;
                     }
-                    total -= *p;
                 }
                 Err(Box::new(Error::InternalError(
                     "miscalculation of total hybrid probabilities".to_string(),
                 )))
             }
+            Workload::Delay { after: _, wrap } => Box::pin(wrap.step(client, data_set, guac)).await,
+        }
+    }
+
+    pub fn is_active(&self) -> bool {
+        match self {
+            Workload::Nop => true,
+            Workload::ByName(_) => true,
+            Workload::Get(_) => true,
+            Workload::Query(_) => true,
+            Workload::Hybrid(hybrid) => hybrid.iter().any(|(_, w)| w.is_active()),
+            Workload::Delay { after, wrap } => chrono::Utc::now() >= *after && wrap.is_active(),
```

Comment:
From my understanding the delay workload will be run after a specific timestamp. This seems to imply that the same workflow config can only be used once (or some other script have to generate the config with a timestamp each time it run). May I ask why this is designed like this?

Reply:
It was largely an artifact of not thinking about my types.  Will update.